nemo/collections/llm/modelopt/quantization/quantizer.py

```diff
@@ -480,12 +506,17 @@ def export_hf_checkpoint(
         if model is None:
             model, _ = exporter.nemo_load(model_dir)
         unwrapped_model = unwrap_for_modelopt_operations(model)
-        if not mto.ModeloptStateManager.is_converted(unwrapped_model):
-            return None  # Model was not converted by ModelOpt.
 
         with torch.inference_mode():
             with tempfile.TemporaryDirectory() as tmp_dir:
                 exporter.config.save_pretrained(tmp_dir)
+                # For llama4, we only deal with the language_model, vision_model and multi_modal_projector will be acquired from the huggingface checkpoint
+                if (
+                    hasattr(unwrapped_model, 'language_model')
+                    and hasattr(unwrapped_model, 'vision_model')
+                    and hf_checkpoint
+                ):
+                    tmp_dir = hf_checkpoint
```

Comment:
I guess `exporter.config.save_pretrained(tmp_dir)` is not needed in Llama4 case so lets move it to `else` section.

Also, for VLMs, we should `assert hf_checkpoint is not None`

```suggestion
                # For llama4, we only deal with the language_model. vision_model and multi_modal_projector will be acquired from the huggingface checkpoint
                if (
                    hasattr(unwrapped_model, 'language_model')
                    and hasattr(unwrapped_model, 'vision_model')
                ):
                    assert hf_checkpoint is not None, "hf_checkpoint is required for exporting VLMs!"
                    tmp_dir = hf_checkpoint
                else:
                    exporter.config.save_pretrained(tmp_dir)
```

Reply:
Fixed, thanks!